client/blocks/edit-gravatar/index.jsx

```diff
@@ -254,6 +300,7 @@ const recordReceiveImageEvent = () => recordTracksEvent( 'calypso_edit_gravatar_
 export default connect(
 	( state ) => ( {
 		user: getCurrentUser( state ) || {},
+		userSettings: getUserSettings( state ),
```

Comment:
Using the `getUserSetting` selector instead means that the component will update immediately when the user toggles rather than having to wait until the user clicks "Save changes". (that's because this selector consults the unsaved settings too).

It also has the advantage that it will only rerender the component when the setting we're interested in changes, not when _any_ setting changes.

```suggestion
		isGravatarProfileHidden: getUserSetting( state, 'gravatar_profile_hidden' ),
```

Reply:
Just realised you'd still need to use the `getUserSettings()` selector in order to decide whether the settings were currently loading and the placeholders should be rendered. But personally, I'd still use the singular `getUserSetting()` to get the boolean from the store, since it feels much nicer to have the Gravatar component update in real-time as you toggle the setting.